compiler/main/driver.cpp

```diff
@@ -804,9 +805,30 @@ static int invokeChplWithArgs(int argc, char* argv[],
 static int runDriverPhaseOne(int argc, char* argv[]);
 static int runDriverPhaseTwo(int argc, char* argv[]);
 
+// Skip phase two if the compile command does not require it, or if we are
+// running in a debugger for phase one only. By default, warn if skipping
+// for the debugger reason.
+static bool shouldSkipDriverPhaseTwo(bool warnIfSkipping = true) {
+  // Check if skipping due to only debugging phase one.
+  bool debugPhaseOneOnly =
+      (fRungdb || fRunlldb) && (driverDebugPhaseOne && !driverDebugPhaseTwo);
+  if (debugPhaseOneOnly && warnIfSkipping) {
+    USR_WARN(
+        "Skipping phase two due to running only phase one "
```

Comment:
Nit: I would say "Skipping driver phase two...." or  "...compiler driver phase two..." or something along those lines to help make it clearer which phase two. (It might be obvious here, but extra clarity can't hurt)

Reply:
Sounds good, I've gone ahead and rolled that into [the phase rename PR](https://github.com/chapel-lang/chapel/pull/23888) since it includes a lot of changes along those lines